File: app/src/main/java/uci/localproxy/proxyscreen/ProxyContract.java
package uci.localproxy.proxyscreen;

import androidx.annotation.NonNull;

import java.util.List;

import uci.localproxy.BasePresenter;
import uci.localproxy.BaseView;
import uci.localproxy.data.profile.Profile;
import uci.localproxy.data.user.User;

/**
 * Created by daniel on 15/09/17.
 */

public interface ProxyContract {

    interface Presenter extends BasePresenter {

        void startProxyFromFabButton(@NonNull String user, @NonNull String pass, @NonNull String profileID,
                        @NonNull String localPort, @NonNull boolean rememberPass,
                        @NonNull boolean setGlobalProxy);

        void startProxyFromHelpDialog(@NonNull String user, @NonNull String pass, @NonNull String profileID,
                                     @NonNull String localPort, @NonNull boolean rememberPass,
                                     @NonNull boolean setGlobalProxy, @NonNull boolean dontShowAgain);

        void stopCredentialCheckTask();

        void stopProxy();

        void onDestroy();

        void addNewProfile();

        void goToWifiConfDialog();

        void goToWifiSettings(boolean dontShowAgain);

        void filterUsers(String constraint);

        void loadUsers();

        void startServiceResult(int message);
    }

    interface View extends BaseView<Presenter> {

        void enableAllViews();

        void disableAllViews();

        void setPlayView();

        void setStopView();

        void setUsername(String username);

        void setPassword(String password);

        void setRememberPassword(boolean remember);

        void setLocalPort(String localPort);

        void setGlobalProxyChecked(boolean checked);

        void setSpinnerProfiles(List<Profile> profiles);

        void setSpinnerProfileSelected(String profileId);

        void showNoProfilesView();

        void setUsernameEmptyError();

        void setPasswordEmptyError();

        void setProfileNoSelectedError();

        void setLocalPortEmptyError();

        void setLocalPortOutOfRangeError();

        void setLocalPortNotAvailable();

        void setUsers(@NonNull List<User> users);

        void showAddProfile();

//        boolean isProxyServiceRunning();

        boolean isConnectedToAWifi();

        void stopProxyService();

        void startProxyService(String username, String password, String server,
                               int inputport, int outputport, String bypass,
                               String domain, boolean setGlobProxy);

        void showWifiConfDialog();

        void startWifiConfActivity();

        void showProgressDialog(boolean show);

        void showWrongCredentialsDialog();

        void showNetworkError();

        void showConnectionTimeOutError();

        void showUnknownHostError();

        void showConnectionError();

        void showErrorStartingService();

    }
}

File: app/src/main/java/grgr/localproxy/proxyscreen/ProxyContract.kt
package grgr.localproxy.proxyscreen

import grgr.localproxy.BasePresenter
import grgr.localproxy.BaseView
import grgr.localproxy.proxydata.profile.Profile
import grgr.localproxy.proxydata.user.User

/**
 * Created by daniel on 15/09/17.
 */
interface ProxyContract {
    interface Presenter : BasePresenter {
        fun startProxyFromFabButton(
            user: String, pass: String, profileID: String,
            localPort: String, rememberPass: Boolean,
            setGlobalProxy: Boolean
        )

        fun startProxyFromHelpDialog(
            user: String, pass: String, profileID: String,
            localPort: String, rememberPass: Boolean,
            setGlobalProxy: Boolean, dontShowAgain: Boolean
        )

        fun stopCredentialCheckTask()
        fun stopProxy()
        fun onDestroy()
        fun addNewProfile()
        fun goToWifiConfDialog()
        fun goToWifiSettings(dontShowAgain: Boolean)
        fun filterUsers(constraint: String?)
        fun loadUsers()
        fun startServiceResult(message: Int)
    }

    interface View : BaseView<Presenter?> {
        fun enableAllViews()
        fun disableAllViews()
        fun setPlayView()
        fun setStopView()
        fun setUsername(username: String?)
        fun setPassword(password: String?)
        fun setRememberPassword(remember: Boolean)
        fun setLocalPort(localPort: String?)
        fun setGlobalProxyChecked(checked: Boolean)
        fun setSpinnerProfiles(profiles: List<Profile>?)
        fun setSpinnerProfileSelected(profileId: String)
        fun showNoProfilesView()
        fun setUsernameEmptyError()
        fun setPasswordEmptyError()
        fun setProfileNoSelectedError()
        fun setLocalPortEmptyError()
        fun setLocalPortOutOfRangeError()
        fun setLocalPortNotAvailable()
        fun setUsers(users: List<User>)
        fun showAddProfile()

        //        boolean isProxyServiceRunning();
        val isConnectedToAWifi: Boolean

        fun stopProxyService()
        fun startProxyService(
            username: String?, password: String?, server: String?,
            inputport: Int, outputport: Int, bypass: String?,
            domain: String?, setGlobProxy: Boolean
        )

        fun showWifiConfDialog()
        fun startWifiConfActivity()
        fun showProgressDialog(show: Boolean)
        fun showWrongCredentialsDialog()
        fun showNetworkError()
        fun showConnectionTimeOutError()
        fun showUnknownHostError()
        fun showConnectionError()
        fun showErrorStartingService()
    }
}